### PR TITLE
CI: properly test min and dev

### PR DESCRIPTION
**Pull request by martinfleis**
Minor changes to CI environments to ensure that we check against minimal supported versions as well as against dev versions. That ensures that issues like #254 are caught sooner and that we know it affects only some versions of dependencies.

### CODE REVIEW

**Comment by martinfleis** [comment]
This looks as expected now. The envs with old pin pass, latest fail due to scikit-learn regression, dev passes again as that regression was fixed on sklearn main.

I have also updated pins in requirements to reflect the actual minimum (or at least tested minimum).

File: ci/38-minimal-numba.yaml [hunk 1]
```diff
@@ -17,5 +17,5 @@ dependencies:
   - pytest-xdist
   - rtree
   # optional
-  - geopandas>=0.7.0
+  - geopandas>=0.12.0,<0.13
```
Comment (martinfleis) [inline ci/38-minimal-numba.yaml:20]: older pandas does not support shapely 2

**Comment by jGaboardi** [comment]
> This looks as expected now. The envs with old pin pass, latest fail due to scikit-learn regression, dev passes again as that regression was fixed on sklearn main.
> 
> I have also updated pins in requirements to reflect the actual minimum (or at least tested minimum).

@martinfleis is there a specific issue or PR over in `sklearn` we can tag here for sake of thoroughness?

**Comment by martinfleis** [comment]
@jGaboardi I suppose https://github.com/scikit-learn/scikit-learn/pull/26772

**Review by jGaboardi — APPROVED** [review]
LGTM

**Pull request merged**